### PR TITLE
fix: change database save in DatasourceEditor

### PR DESCRIPTION
**Pull request by mistercrunch**
This addresses the issue where pointing a datasource to another database
in the datasource editor is not reflected.

Also addresses:
- a minor cosmetic issue in the datasource editor.
- user/owners list not getting populated

### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -745,6 +745,7 @@ def import_dashboards(self):
             try:
                 dashboard_import_export.import_dashboards(db.session, f.stream)
             except DatabaseNotFound as e:
+                logger.exception(e)
```
Comment (willbarrett) [inline superset/views/core.py]: 👍

File: tests/base_tests.py [hunk 2]
```diff
@@ -268,7 +268,7 @@ def create_fake_db(self):
             cls=models.Database,
             criteria={"database_name": database_name},
             session=db.session,
-            sqlalchemy_uri="sqlite://test",
+            sqlalchemy_uri="sqlite:///:memory:",
```
Comment (willbarrett) [inline tests/base_tests.py]: 👍

File: tests/fixtures/datasource.py [hunk 3]
```diff
@@ -18,6 +18,7 @@
 datasource_post = {
     "id": None,
     "column_formats": {"ratio": ".2%"},
+    "database": {"id": 1},
```
Comment (willbarrett) [inline tests/fixtures/datasource.py]: Eventually it would be good to move these to randomly generated IDs (note to self)

**Review by willbarrett — APPROVED** [review]

File: tests/fixtures/datasource.py (hunk 3 above)
Comment (mistercrunch) [reply tests/fixtures/datasource.py]: Yes, actually here I replace it in the test itself. Not sure what the proper approach here, a function that receives an object and returns a fixture? I'm guessing with DAO abstraction we can get more creative on generating proper fixtures.

**Pull request merged**